packages/gatsby/src/internal-plugins/dev-404-page/raw_dev-404-page.js

```diff
@@ -11,13 +12,16 @@ class Dev404Page extends React.Component {
 
   constructor(props) {
     super(props)
-    const { data } = this.props
+    const { data, location } = this.props
     const pagePaths = data.allSitePage.nodes.map(node => node.path)
+    const urlState = queryString.parse(location.search)
+
+    const initialPagePathSearchTerms = urlState.q ? urlState.q : ``
```

Comment:
Happy to choose any other query string parameter.

Reply:
I think `q` is fine, but see no reason not to use more meaningful name - like `path-filter` (or even just `filter`). With code that auto updates search params in browser's address bar, there's not much reason not to

Reply:
I'll change it to filter then